python/whylogs/core/dataset_profile.py

```diff
@@ -108,7 +108,9 @@ def _do_track(
         # TODO: do this less frequently when operating at row level
         dirty = self._schema.resolve(pandas=pandas, row=row)
         if dirty:
-            new_cols = self._schema.get_col_names().difference(self._columns.keys())
+            schema_col_keys = self._schema.get_col_names()
+            col_keys = tuple(self._columns.keys())
```

Comment:
Actually intead of changing the tips, you can change up lines 112 and 113 to be able to use a hash more optimally for the difference using by using this:

`new_cols = (col for col in schema_col_keys if col not in self._columns)`

Reply:
yes, that's much better!
Thanks, Melly!
changed in the last commit!